discovery/utils/file_handler.py

```diff
@@ -37,15 +41,38 @@ def load_file(self, file_path):
         if extension not in self.supported_extensions:
             raise UnsupportedFileExtension(file_path)
 
+        if self._should_file_be_ignored(file_path):
+            return
         # Nothing in place to prevent reloading files for now
         handler = self.supported_extensions[extension]
         logger.debug(f"Loading file {file_path} using \"{handler.__name__}\" handler")
         self.loaded_files[file_path] = handler(file_path)
 
+    def _should_file_be_ignored(self, path):
+        return any(path.endswith(ignored_extension) for ignored_extension in self.ignored_extensions)
+
     @staticmethod
     def _handle_csv(file_path):
-        return pandas.read_csv(file_path)
+        dataframe = pandas.read_csv(file_path)
+        return FileHandler.construct_file_descriptor(file_path, FileExtension.CSV, dataframe)
 
     @staticmethod
     def _handle_json(file_path):
-        return pandas.read_json(file_path)
\ No newline at end of file
+        dataframe = pandas.read_json(file_path)
+        return FileHandler.construct_file_descriptor(file_path, FileExtension.JSON, dataframe)
+
+    @staticmethod
+    def construct_file_descriptor(file_path: str, extension: FileExtension, dataframe: pandas.DataFrame):
```

Comment:
this might be better as a function
to avoid confusion its also possible to call this by using:
self.__class__.construct_file_descriptor(*args, **kwargs)
rather than specifying the static class each time

Reply:
This view doesn't let me see updates to the code, it might be better to just screenshot to discord and then we'll chat about whether its resolved